Marlin/src/inc/Conditionals_post.h

```diff
@@ -2946,7 +2946,11 @@
 /**
  * Bed Probe dependencies
  */
-#if ANY(MESH_BED_LEVELING, HAS_BED_PROBE)
+#if ANY(BABYSTEPPING, PROBE_SELECTED)
+  #define HAS_ZOFFSET_ITEM 1
+#endif
+
+#if HAS_ZOFFSET_ITEM
   #ifndef PROBE_OFFSET_ZMIN
     #define PROBE_OFFSET_ZMIN -20
```

Comment:
These value range settings should only pertain to editing of the Probe Z Offset. Note that a UI may include both a baby-stepping item that has no effect on the probe offset, and a Probe Z Offset adjustment that doesn't baby-step. Non-probe-offset-adjusting baby-stepping not limited to any range.

Reply:
the main concern is `DWIN_LCD_PROUI` uses `PROBE_OFFSET_ZMIN/MAX` whether it `HAS_MESH`, a probe --auto/mesh bed leveling, or none of the above. because it uses that offset value to adjust the Z-Offset (nozzle to bed relation).

I mean if this **PR** change isn't good, we could always just redefine it in **dwin_defines.h** into something else like  
```y
#ifndef PROBE_Z_OFFSET_ZMIN
  #define Z_OFFSET_ZMIN   20
#else
  #define Z_OFFSET_ZMIN  PROBE_Z_OFFSET_ZMIN
#endif
```
and use the new value in for the max/min of the function

Reply:
[As I mentioned above](https://github.com/MarlinFirmware/Marlin/pull/26502#issuecomment-1846055677), these UIs need to decouple from requiring/assuming a probe is installed so we don't end up with these kind of bug reports when someone doesn't have specific hardware or enable a specific feature:

- https://github.com/MarlinFirmware/Marlin/issues/26505